core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -241,8 +240,8 @@ protected void tryComplete() {
 
   protected synchronized void tryTerminate() {
     if (this.state == STATE_COMPLETE && (externalCompletion == null || externalCompletion.isDone())) {
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug(this + " terminated.");
+      if (debugLogEnabled) {
+        LOGGER.debug("{} terminated.", this);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Invoke method(s) only conditionally. ([squid:S2629](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2629))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7701&issues=AWkLjRWWotIXnmsUmgk5&open=AWkLjRWWotIXnmsUmgk5)

[ISSUE_KEY]: AWkLjRWWotIXnmsUmgk5